### DISCUSSION TITLE
dt suddenly reduces to 1e-13 and program stops working. How can I fix this issue?

### DISCUSSION THREAD

**Discussion by K-Komal-98** [idaholab/moose#24232] [Q&A Tools] [answered]
I am working on a dendrite growth problem in solid state electrolyte.
There are two input files:

### without mechanics. 
This works well even beyond 17000 time steps
### with mechanics. 
The input file is exactly same as the above one with just the tensor mechanics included in it. I have consistently used the normalization parameters. This works well upto 169 time steps and suddenly dt reduces to 1e-13. 

I have followed the instructions given in this link but couldn't solve the issue: https://mooseframework.inl.gov/application_usage/failed_solves.html

I am attaching the input files for both the parts and marked the added part in between --------------
Input file WITHOUT MECHANICS:
[input_without_mechanics.txt](https://github.com/idaholab/moose/files/11362155/input_without_mechanics.txt)
Input file WITH MECHANICS:
[input_with_mechanics.txt](https://github.com/idaholab/moose/files/11362153/input_with_mechanics.txt)

This is my thesis work. Can you please guide me what can I do to solve this issue?

**Comment by GiudGiud** [comment]
Hello

Can you reduce the problem down to <5k dofs and use these preconditioning options:
```
  petsc_options = '-pc_svd_monitor'
  petsc_options_iname = '-pc_type'
  petsc_options_value = 'svd'
```
this will tell us more about the conditioning of the matrix

Guillaume

**Reply by K-Komal-98** [reply]
Hi Guillaume,
Thanks for the suggestions. 
I gave 2 runs on hpc with 40 cores:
1. Reduced the dofs to 13000 by reducing the number of points. The program worked for 65 time steps and then reduces the time step until 1e-13.
2. Reduced the dofs to 13000 and used the preconditioning as per your instructions. It has been more than 40 mins. It seems like linear solve is taking too long to solve. Here is the attached file.
[test.txt](https://github.com/idaholab/moose/files/11362733/test.txt)

Thanks

**Reply by GiudGiud** [reply]
13k is still too many. 
5k is almost a hard maximum, we usually advise 1k dofs for SVD

**Reply by K-Komal-98** [reply]
Now, the dofs are less than 5000. It stopped running after 3 iterations. Here is the attached file. 
[test.txt](https://github.com/idaholab/moose/files/11362986/test.txt)

**Reply by GiudGiud** [reply]
Ok no insights for this log unfortunately.
Can you please attach the log for the original large failing case as well?

**Comment by K-Komal-98** [comment]
Here is the log [test.txt](https://github.com/idaholab/moose/files/11363166/test.txt)
Thanks!

**Reply by GiudGiud** [reply]
The convergence history seems quite rough. But it makes progress.
When it goes bad, you get `  Linear solve did not converge due to DIVERGED_NANORINF iterations 0`
this makes me think one of the models in the calculation is returning a NaN

You could compile in debug mode and add run with --trap-fpe on the command line. This may be able to find where it's happening

**Comment by K-Komal-98** [comment]
This is the command line I gave. Can you confirm if it's the correct way to write as it runs upto 2 iterations and then ends.

mpirun -machinefile $PBS_NODEFILE -n 40 /home/mkankani21/offline/nucleation_bm/nucleation_bm-opt --trap-fpe -i input_var_dt.i> test.txt

**Reply by GiudGiud** [reply]
Does it end with a trapping of a floating point exception? Or with nothing useful?

It's correct except you should use a debug executable, not the opt one

**Reply by K-Komal-98** [reply]
I will make the debug executable and share the results. One doubt I have is that while making the debug executable, by any chance does it modify other files?

**Comment by K-Komal-98** [comment]
The system runs for 2 iterations and this is what I get. Am I missing something?
```
Floating point exception signaled (
Floating point exception signaled (invalid floating point operation)!

To track this down, compile in debug mode, then in gdb do:
  break libmesh_handleFPE
  run ...
  bt

Floating point exception signaled (invalid floating point operation)!

To track this down, compile in debug mode, then in gdb do:
  break libmesh_handleFPE
  run ...
  bt
invalid floating point operation)!

To track this down, compile in debug mode, then in gdb do:
  break libmesh_handleFPE
  run ...
  bt
[25] ../src/base/libmesh.C, line 134, compiled Oct  7 2022 at 16:28:19
[29] ../src/base/libmesh.C, line 134, compiled Oct  7 2022 at 16:28:19
[36] ../src/base/libmesh.C, line 134, compiled Oct  7 2022 at 16:28:19

Floating point exception signaled (invalid floating point operation)!

To track this down, compile in debug mode, then in gdb do:
  break libmesh_handleFPE
  run ...
  bt
[39] ../src/base/libmesh.C, line 134, compiled Oct  7 2022 at 16:28:19

Floating point exception signaled (invalid floating point operation)!

To track this down, compile in debug mode, then in gdb do:
  break libmesh_handleFPE
  run ...
  bt
[37] ../src/base/libmesh.C, line 134, compiled Oct  7 2022 at 16:28:19

Floating point exception signaled (invalid floating point operation)!

To track this down, compile in debug mode, then in gdb do:
  break libmesh_handleFPE
  run ...
  bt
[28] ../src/base/libmesh.C, line 134, compiled Oct  7 2022 at 16:28:19

Floating point exception signaled (invalid floating point operation)!

To track this down, compile in debug mode, then in gdb do:
  break libmesh_handleFPE
  run ...
  bt
[38] ../src/base/libmesh.C, line 134, compiled Oct  7 2022 at 16:28:19
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 25
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 28
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 29
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 36
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 37
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 38
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 39
[16:hpc857] unexpected disconnect completion event from [28:hpc858]
Assertion failed in file ../../src/mpid/ch3/channels/nemesis/netmod/dapl/dapl_conn_rc.c at line 1182: 0
internal ABORT - process 16
[17:hpc857] unexpected disconnect completion event from [28:hpc858]
Assertion failed in file ../../src/mpid/ch3/channels/nemesis/netmod/dapl/dapl_conn_rc.c at line 1182: 0
internal ABORT - process 17
[18:hpc857] unexpected disconnect completion event from [28:hpc858]
Assertion failed in file ../../src/mpid/ch3/channels/nemesis/netmod/dapl/dapl_conn_rc.c at line 1182: 0
internal ABORT - process 18
[19:hpc857] unexpected disconnect completion event from [28:hpc858]
Assertion failed in file ../../src/mpid/ch3/channels/nemesis/netmod/dapl/dapl_conn_rc.c at line 1182: 0
internal ABORT - process 19
[9:hpc857] unexpected disconnect completion event from [28:hpc858]
Assertion failed in file ../../src/mpid/ch3/channels/nemesis/netmod/dapl/dapl_conn_rc.c at line 1182: 0
internal ABORT - process 9
[11:hpc857] unexpected disconnect completion event from [28:hpc858]
Assertion failed in file ../../src/mpid/ch3/channels/nemesis/netmod/dapl/dapl_conn_rc.c at line 1182: 0
internal ABORT - process 11
[13:hpc857] unexpected disconnect completion event from [28:hpc858]
Assertion failed in file ../../src/mpid/ch3/channels/nemesis/netmod/dapl/dapl_conn_rc.c at line 1182: 0
internal ABORT - process 13
[15:hpc857] unexpected disconnect completion event from [28:hpc858]
Assertion failed in file ../../src/mpid/ch3/channels/nemesis/netmod/dapl/dapl_conn_rc.c at line 1182: 0
internal ABORT - process 15
```

**Reply by GiudGiud** [reply]
so trap-fpe doesnt catch it but it s telling you to catch it in a debugger. 
please follow the instructions to do it

**Reply by GiudGiud** [reply]
if you're not familiar with debugging, here's a good start
https://mooseframework.inl.gov/application_development/debugging.html

**Comment by K-Komal-98** [comment]
I followed the instructions. I am attaching the file. Let me know if I am in the right direction.
[debug.txt](https://github.com/idaholab/moose/files/11376570/debug.txt)

**Reply by GiudGiud** [reply]
add this breakpoint too `break libmesh_handleFPE`

also once it breaks, please post the backtrace

**Comment by K-Komal-98** [comment]
Do I have to just enter this command in gdb mode or something else also has to be done? I followed the previous steps and I am not sure if it is working differently.

This is what I have done, it is debugging

(gdb) break libmesh_handleFPE
Function "libmesh_handleFPE" not defined.
Make breakpoint pending on future shared library load? (y or [n]) y

Breakpoint 2 (libmesh_handleFPE) pending.
(gdb) r
The program being debugged has been started already.
Start it from the beginning? (y or n) y

Starting program: /home/mkankani21/offline/nucleation_bm/./nucleation_bm-dbg -i input_var_dt.i

**Reply by GiudGiud** [reply]
this should be fine

**Comment by K-Komal-98** [comment]
Here it is:

Time Step 1, time = 1, dt = 1

    Computing Initial Residual
      Computing Residual......                                                           [ 36.69 s] [    9 MB]
    Finished Computing Initial Residual                                                  [ 36.69 s] [    9 MB]
    Computing Residual......                                                             [ 38.56 s] [  -10 MB]
 0 Nonlinear |R| = 3.239109e+06
    Computing Jacobian.................                                                  [ 92.32 s] [   64 MB]
      0 Linear |R| = 3.239109e+06
      1 Linear |R| = 4.968937e-08

    Computing Residual......                                                             [ 36.71 s] [    4 MB]
 1 Nonlinear |R| = 6.198875e+03
    Computing Jacobian................                                                   [ 86.68 s] [    0 MB]
      0 Linear |R| = 6.198875e+03
      1 Linear |R| = 1.008331e-09

Program received signal SIGFPE, Arithmetic exception.
FunctionParserBase<double>::Eval (this=0x36e0f10, Vars=0x36bd590)
    at ../../../contrib/fparser/fparser.cc:2985
2985	              Stack[SP-1] = Stack[SP] / Stack[SP-1]; --SP; break;

**Reply by GiudGiud** [reply]
do you get a longer backtrace? with `bt`

**Reply by K-Komal-98** [reply]
Yes. Here it is:

(gdb) bt
#0  FunctionParserBase<double>::Eval (this=0x36e0f10, Vars=0x36bd590)
    at ../../../contrib/fparser/fparser.cc:2985
#1  0x00007fffedf37b4a in FunctionParserADBase<double>::Eval (this=0x36e0f10, 
    Vars=0x36bd590) at ../../../contrib/fparser/fparser_ad.cc:644
#2  0x00007ffff3367da3 in FunctionParserUtils<false>::evaluate (
    this=0x36dd6f0, parser=..., name=...)
    at /home/mkankani21/offline/moose/framework/src/utils/FunctionParserUtils.C:102
#3  0x00007ffff2bbdc1e in DerivativeParsedMaterialHelperTempl<false>::computeQpProperties (this=0x36dc6b0)
    at /home/mkankani21/offline/moose/framework/src/materials/DerivativeParsedMaterialHelper.C:286
#4  0x00007ffff2ba3974 in Material::computeProperties (this=0x36dc6b0)
    at /home/mkankani21/offline/moose/framework/src/materials/Material.C:132
#5  0x00007ffff2fcf106 in MaterialData::reinit<std::vector<std::shared_ptr<MaterialBase>, std::allocator<std::shared_ptr<MaterialBase> > > > (this=0x31456e0, 
    mats=...)
    at /home/mkankani21/offline/moose/framework/build/header_symlinks/MaterialData.h:458
#6  0x00007ffff2f8bb14 in FEProblemBase::reinitMaterials (this=0x312d250, 
    blk_id=0, tid=0, swap_stateful=true)
    at /home/mkankani21/offline/moose/framework/src/problems/FEProblemBase.C:3246

**Reply by GiudGiud** [reply]
there s your problem

one of your derivativeParsedMaterial has a floating point exception, probably a division by 0
~~can you paste them here?~~ -> I went to have a look. Not super clear which one, any luck for you?

**Reply by GiudGiud** [reply]
actually if you are still in the debbuger, it would really help to find the name of the material

**Reply by K-Komal-98** [reply]
yes, I am still in the debugger. What shall I do next?

**Reply by GiudGiud** [reply]
info local (or locals?) with gdb
or fr v with lldb

**Reply by GiudGiud** [reply]
or maybe `b  Material::computeProperties`
then continue
on the next break `name()`

hopefully still the same material that is buggy

**Reply by K-Komal-98** [reply]
There were still more. From which line did you find out the division error?
```
(gdb) bt
#0  FunctionParserBase<double>::Eval (this=0x36e0f10, Vars=0x36bd590)
    at ../../../contrib/fparser/fparser.cc:2985
#1  0x00007fffedf37b4a in FunctionParserADBase<double>::Eval (this=0x36e0f10, 
    Vars=0x36bd590) at ../../../contrib/fparser/fparser_ad.cc:644
#2  0x00007ffff3367da3 in FunctionParserUtils<false>::evaluate (
    this=0x36dd6f0, parser=..., name=...)
    at /home/mkankani21/offline/moose/framework/src/utils/FunctionParserUtils.C:102
#3  0x00007ffff2bbdc1e in DerivativeParsedMaterialHelperTempl<false>::computeQpProperties (this=0x36dc6b0)
    at /home/mkankani21/offline/moose/framework/src/materials/DerivativeParsedMaterialHelper.C:286
#4  0x00007ffff2ba3974 in Material::computeProperties (this=0x36dc6b0)
    at /home/mkankani21/offline/moose/framework/src/materials/Material.C:132
#5  0x00007ffff2fcf106 in MaterialData::reinit<std::vector<std::shared_ptr<MaterialBase>, std::allocator<std::shared_ptr<MaterialBase> > > > (this=0x31456e0, 
    mats=...)
    at /home/mkankani21/offline/moose/framework/build/header_symlinks/MaterialData.h:458
#6  0x00007ffff2f8bb14 in FEProblemBase::reinitMaterials (this=0x312d250, 
    blk_id=0, tid=0, swap_stateful=true)
    at /home/mkankani21/offline/moose/framework/src/problems/FEProblemBase.C:3246
---Type <return> to continue, or q <return> to quit---return
#7  0x00007ffff26b4b49 in ComputeResidualThread::onElement (
    this=0x7fffffffa7d0, elem=0x1189690)
    at /home/mkankani21/offline/moose/framework/src/loops/ComputeResidualThread.C:144
#8  0x00007ffff2707976 in ThreadedElementLoopBase<libMesh::StoredRange<libMesh::MeshBase::const_element_iterator, libMesh::Elem const*> >::operator() (
    this=0x7fffffffa7d0, range=..., bypass_threading=false)
    at /home/mkankani21/offline/moose/framework/build/header_symlinks/ThreadedElementLoopBase.h:226
#9  0x00007ffff2547595 in libMesh::Threads::parallel_reduce<libMesh::StoredRange<libMesh::MeshBase::const_element_iterator, libMesh::Elem const*>, ComputeResidualThread> (range=..., body=...)
    at /home/mkankani21/offline/moose/scripts/../libmesh/installed/include/libmesh/threads_pthread.h:380
#10 0x00007ffff2515188 in NonlinearSystemBase::computeResidualInternal (
    this=0x3145b90, tags=...)
    at /home/mkankani21/offline/moose/framework/src/systems/NonlinearSystemBase.C:1483
#11 0x00007ffff250fba3 in NonlinearSystemBase::computeResidualTags (
    this=0x3145b90, tags=...)
    at /home/mkankani21/offline/moose/framework/src/systems/NonlinearSystemBase.C:742
#12 0x00007ffff2fa10fd in FEProblemBase::computeResidualTags (this=0x312d250, 
---Type <return> to continue, or q <return> to quit---return
    tags=...)
    at /home/mkankani21/offline/moose/framework/src/problems/FEProblemBase.C:5626
#13 0x00007ffff2f9fcb2 in FEProblemBase::computeResidualInternal (
    this=0x312d250, soln=..., residual=..., tags=...)
    at /home/mkankani21/offline/moose/framework/src/problems/FEProblemBase.C:5464
#14 0x00007ffff2f9f760 in FEProblemBase::computeResidual (this=0x312d250, 
    soln=..., residual=...)
    at /home/mkankani21/offline/moose/framework/src/problems/FEProblemBase.C:5419
#15 0x00007ffff2f9f54d in FEProblemBase::computeResidualSys (this=0x312d250, 
    soln=..., residual=...)
    at /home/mkankani21/offline/moose/framework/src/problems/FEProblemBase.C:5394
#16 0x00007ffff25055fb in ComputeResidualFunctor::residual (this=0x3147358, 
    soln=..., residual=..., sys=...)
    at /home/mkankani21/offline/moose/framework/src/systems/ComputeResidualFunctor.C:23
#17 0x00007fffedb3c5f6 in libMesh::libmesh_petsc_snes_residual (
    snes=0x4ef2f00, x=0x42a7b40, r=0x42a8220, ctx=0x3148360)
    at ../src/solvers/petsc_nonlinear_solver.C:163
#18 0x00007fffe9e3400e in SNESComputeFunction (snes=0x4ef2f00, x=0x42a7b40, 
---Type <return> to continue, or q <return> to quit---return
    y=0x42a8220)
    at /home/mkankani21/offline/moose/petsc/src/snes/interface/snes.c:2471
#19 0x00007fffe9e573f8 in SNESLineSearchApply_BT (linesearch=0x4294d00)
    at /home/mkankani21/offline/moose/petsc/src/snes/linesearch/impls/bt/linesearchbt.c:147
#20 0x00007fffe9e5d54c in SNESLineSearchApply (linesearch=0x4294d00, 
    X=X@entry=0x32aca70, F=F@entry=0x7c8ad0, fnorm=fnorm@entry=0x7fffffffc2e8, 
    Y=Y@entry=0x4eecd50)
    at /home/mkankani21/offline/moose/petsc/src/snes/linesearch/interface/linesearch.c:619
#21 0x00007fffe9e77125 in SNESSolve_NEWTONLS (snes=0x4ef2f00)
    at /home/mkankani21/offline/moose/petsc/src/snes/impls/ls/ls.c:239
#22 0x00007fffe9e44987 in SNESSolve (snes=0x4ef2f00, b=0x0, x=<optimized out>)
    at /home/mkankani21/offline/moose/petsc/src/snes/interface/snes.c:4809
#23 0x00007fffedb404a9 in libMesh::PetscNonlinearSolver<double>::solve (
    this=0x3148360, pre_in=..., x_in=..., r_in=...)
    at ../src/solvers/petsc_nonlinear_solver.C:976
#24 0x00007fffedbcaa83 in libMesh::NonlinearImplicitSystem::solve (
    this=0x3147390) at ../src/systems/nonlinear_implicit_system.C:191
#25 0x00007ffff1fea642 in TimeIntegrator::solve (this=0x32f0010)
    at /home/mkankani21/offline/moose/framework/src/timeintegrators/TimeIntegrator.C:54
#26 0x00007ffff2509bce in NonlinearSystem::solve (this=0x3145b90)
---Type <return> to continue, or q <return> to quit---return
    at /home/mkankani21/offline/moose/framework/src/systems/NonlinearSystem.C:195
#27 0x00007ffff2f9bf74 in FEProblemBase::solve (this=0x312d250)
    at /home/mkankani21/offline/moose/framework/src/problems/FEProblemBase.C:5131
#28 0x00007ffff27173b2 in FEProblemSolve::solve (this=0x32e3160)
    at /home/mkankani21/offline/moose/framework/src/executioners/FEProblemSolve.C:253
#29 0x00007ffff271b8f7 in FixedPointSolve::solveStep (this=0x32e3540, 
    begin_norm=@0x24ce250: 0, end_norm=@0x2007900: 0, transformed_dofs=...)
    at /home/mkankani21/offline/moose/framework/src/executioners/FixedPointSolve.C:393
#30 0x00007ffff271ad0c in FixedPointSolve::solve (this=0x32e3540)
    at /home/mkankani21/offline/moose/framework/src/executioners/FixedPointSolve.C:274
#31 0x00007ffff28d88be in TimeStepper::step (this=0x36abce0)
    at /home/mkankani21/offline/moose/framework/src/timesteppers/TimeStepper.C:161
#32 0x00007ffff2729acd in Transient::takeStep (this=0x32e2e70, input_dt=-1)
    at /home/mkankani21/offline/moose/framework/src/executioners/Transient.C:444
#33 0x00007ffff2728e17 in Transient::execute (this=0x32e2e70)
    at /home/mkankani21/offline/moose/framework/src/executioners/Transient.C:329---Type <return> to continue, or q <return> to quit---return

#34 0x00007ffff329236b in MooseApp::executeExecutioner (this=0x80b4b0)
    at /home/mkankani21/offline/moose/framework/src/base/MooseApp.C:1095
#35 0x00007ffff3294f9a in MooseApp::run (this=0x80b4b0)
    at /home/mkankani21/offline/moose/framework/src/base/MooseApp.C:1410
#36 0x000000000040a010 in main (argc=3, argv=0x7fffffffe2b8)
    at /home/mkankani21/offline/nucleation_bm/src/main.C:33
```

**Reply by GiudGiud** [reply]
I dont know that it s a division error. I know it s a floating point exception.
It s hitting the floating point exception breakpoint, that's how I know

**Comment by K-Komal-98** [comment]
I gave two commands:
```
info local
b  Material::computeProperties

Am I missing something?
(gdb) info local
byteCode = 0x36e30d0
immed = 0x36ddbf0
byteCodeSize = 55
IP = 18
DP = 1
SP = 5
AutoDeallocStack = {ptr = 0x42aa7f0}
Stack = @0x7fffffffa1c0: 0x42aa7f0
(gdb) b  Material::computeProperties
Breakpoint 3 at 0x7ffff2ba37cf: file /home/mkankani21/offline/moose/framework/src/materials/Material.C, line 106.
```

**Reply by GiudGiud** [reply]
you can press `c` to continue to the next breakpoint

**Reply by GiudGiud** [reply]
we want the name of the material causing the FPE (floating point exception) so you can know where to look

**Reply by K-Komal-98** [reply]
Entered c

(gdb) c
Continuing.

    Computing Residual
Floating point exception signaled (invalid floating point operation)!

To track this down, compile in debug mode, then in gdb do:
  break libmesh_handleFPE
  run ...
  bt
Stack frames: 40
0: libMesh::print_trace(std::ostream&)
1: libMesh::MacroFunctions::report_error(char const*, int, char const*, char const*)
2: /home/mkankani21/offline/moose/scripts/../libmesh/installed/lib/libmesh_dbg.so.0(+0x15a728f) [0x7fffecfb528f]
3: /lib64/libpthread.so.0(+0xf100) [0x7fffe4ed5100]
4: FunctionParserBase<double>::Eval(double const*)
5: FunctionParserADBase<double>::Eval(double const*)
6: FunctionParserUtils<false>::evaluate(std::shared_ptr<FunctionParserADBase<double> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
7: DerivativeParsedMaterialHelperTempl<false>::computeQpProperties()
8: Material::computeProperties()
9: void MaterialData::reinit<std::vector<std::shared_ptr<MaterialBase>, std::allocator<std::shared_ptr<MaterialBase> > > >(std::vector<std::shared_ptr<MaterialBase>, std::allocator<std::shared_ptr<MaterialBase> > > const&)
10: FEProblemBase::reinitMaterials(unsigned short, unsigned int, bool)
11: ComputeResidualThread::onElement(libMesh::Elem const*)
12: ThreadedElementLoopBase<libMesh::StoredRange<libMesh::MeshBase::const_element_iterator, libMesh::Elem const*> >::operator()(libMesh::StoredRange<libMesh::MeshBase::const_element_iterator, libMesh::Elem const*> const&, bool)
13: void libMesh::Threads::parallel_reduce<libMesh::StoredRange<libMesh::MeshBase::const_element_iterator, libMesh::Elem const*>, ComputeResidualThread>(libMesh::StoredRange<libMesh::MeshBase::const_element_iterator, libMesh::Elem const*> const&, ComputeResidualThread&)
14: NonlinearSystemBase::computeResidualInternal(std::set<unsigned int, std::less<unsigned int>, std::allocator<unsigned int> > const&)
15: NonlinearSystemBase::computeResidualTags(std::set<unsigned int, std::less<unsigned int>, std::allocator<unsigned int> > const&)
16: FEProblemBase::computeResidualTags(std::set<unsigned int, std::less<unsigned int>, std::allocator<unsigned int> > const&)
17: FEProblemBase::computeResidualInternal(libMesh::NumericVector<double> const&, libMesh::NumericVector<double>&, std::set<unsigned int, std::less<unsigned int>, std::allocator<unsigned int> > const&)
18: FEProblemBase::computeResidual(libMesh::NumericVector<double> const&, libMesh::NumericVector<double>&)
19: FEProblemBase::computeResidualSys(libMesh::NonlinearImplicitSystem&, libMesh::NumericVector<double> const&, libMesh::NumericVector<double>&)
20: ComputeResidualFunctor::residual(libMesh::NumericVector<double> const&, libMesh::NumericVector<double>&, libMesh::NonlinearImplicitSystem&)
21: libmesh_petsc_snes_residual
22: SNESComputeFunction
23: /home/mkankani21/offline/moose/petsc/arch-moose/lib/libpetsc.so.3.16(+0xd2d3f8) [0x7fffe9e573f8]
24: SNESLineSearchApply
25: /home/mkankani21/offline/moose/petsc/arch-moose/lib/libpetsc.so.3.16(+0xd4d125) [0x7fffe9e77125]
26: SNESSolve
27: libMesh::PetscNonlinearSolver<double>::solve(libMesh::SparseMatrix<double>&, libMesh::NumericVector<double>&, libMesh::NumericVector<double>&, double, unsigned int)
28: libMesh::NonlinearImplicitSystem::solve()
29: TimeIntegrator::solve()
30: NonlinearSystem::solve()
31: FEProblemBase::solve()
32: FEProblemSolve::solve()
33: FixedPointSolve::solveStep(double&, double&, std::set<unsigned long, std::less<unsigned long>, std::allocator<unsigned long> > const&)
34: FixedPointSolve::solve()
35: TimeStepper::step()
36: Transient::takeStep(double)
37: Transient::execute()
38: MooseApp::executeExecutioner()
39: MooseApp::run()
.[0] ../src/base/libmesh.C, line 134, compiled Oct  7 2022 at 16:28:19

Breakpoint 1, 0x00007fffe5cd47f0 in PMPI_Abort ()
   from /home/gmaurya20/local/mpich-3.4.2/mpich/lib/libmpi.so.12

**Reply by GiudGiud** [reply, marked as answer]
ok no luck. Well we know one DerivativeParsedMaterial is out of bounds / causing some FPE issues. You ll have to look at them and see which ones that would be

**Reply by K-Komal-98** [reply]
Thankyou for your patience! I am just few steps away. I would request you to help me in what should I fix.
These are the two DerivativeParsedMaterial blocks that I have added in the existing problem that works fine. Probably eta is causing the problem as it ranges from 0 to 1. Is that causing a problem?
```
[./one_minus_h_eta]
    type = DerivativeParsedMaterial
    f_name = one_minus_h
    args = 'eta'
    function = '1-h'
    material_property_names = 'h'
  [../]
  [./var_dependence]
    type = DerivativeParsedMaterial
    f_name = var_dep
    args = 'eta'
    function = eta
    derivative_order = 1
  [../]
```

**Reply by GiudGiud** [reply]
these functions look fine actually.
Unless eta or h is NaN which would mean the variable value has a Nan
